lib/rack/jwt/auth.rb

```diff
@@ -152,22 +152,31 @@ def check_exclude_type!
         end
 
         @exclude.each do |x|
-          unless x.is_a?(String)
-            raise ArgumentError, 'each exclude Array element must be a String'
+          unless x.is_a?(String) || x.is_a?(Regexp)
+            raise ArgumentError, 'each exclude Array element must be a String or a Regexp'
           end
 
-          if x.empty?
+          if x.to_s.empty?
             raise ArgumentError, 'each exclude Array element must not be empty'
           end
 
-          unless x.start_with?('/')
+          # Perhaps surprisingly, Regexp#inspect actually produces the more
+          # natural version of the string than #to_s.
+          as_s = x.is_a?(Regexp) ? x.inspect : x
```

Comment:
I'm not entirely sure about this. `%r{stati+}` turns into `/stati+` so this always passes for Regexp. Which is probably not what we want. We could check for `/\/stati` in case of a regexp, but it becomes icky quite fast.

Do we want to check if it starts with a / for regexp at all?